hal/network/ncp/cellular/cellular_hal.cpp

```diff
@@ -41,6 +41,8 @@ namespace {
 
 using namespace particle;
 
+static CellularDevice cachedInfo = {};
```

Comment:
Not sure if there is a better place to cache this. But this simple implementation works for now